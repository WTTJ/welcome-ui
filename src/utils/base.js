import { createGlobalStyle, css } from '@xstyled/styled-components'

import { reset } from './reset'
import { fontFace } from './font'

const baseResponsiveStyles = css`
  @media (max-width: 1200px) {
    html {
      font-size: 14px;
    }
  }

  @media (max-width: 1300px) and (max-height: 700px) {
    html {
      font-size: 14px;
    }
  }

  @media (max-width: 600px) {
    html {
      font-size: 16px;
    }
  }
`

export const getBaseStyles = theme => createGlobalStyle`
  ${reset}

  ${fontFace(theme)}

  html {
    font-size: ${theme.fontSizes.html};
  }

  body, button, input, select, textarea {
    font-family: ${[theme.fonts.texts, 'sans-serif'].join(', ')};
    -webkit-font-smoothing: antialiased;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${[theme.fonts.headings, theme.fonts.texts, 'serif'].join(', ')};
  }

  h1, h2, h3{
    line-height: 1.3;
  }

  h4, h5, h6, p, li{
    line-height: 1.4;
  }

  ::selection {
    background-color: ${theme.colors.primary[500]};
    color: ${theme.colors.light[200]};
  }

  ${baseResponsiveStyles}
`
