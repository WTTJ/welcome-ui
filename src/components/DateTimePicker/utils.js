export const getDate = (date, interval) => {
  if (!date) {
    return null
  }

  // If invalid date, use today
  let newDate = new Date(date)
  if (isNaN(newDate.getTime())) {
    newDate = new Date()
  }

  // Round to nearest interval
  const nextInterval = Math.ceil(newDate.getMinutes() / interval) * interval
  // Set minutes to nearest interval
  return new Date(newDate.setMinutes(nextInterval, 0, 0))
}
