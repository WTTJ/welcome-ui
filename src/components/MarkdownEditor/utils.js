// Taken from https://github.com/sparksuite/simplemde-markdown-editor/blob/6abda7ab68cc20f4aca870eb243747951b90ab04/src/js/simplemde.js#L140
export const getCurrentToolsFromEditor = editor => {
  const pos = editor.getCursor('start')
  const token = editor.getTokenAt(pos)
  const line = editor.getLine(pos.line)

  const CODE_MIRROR_TYPES = {
    strong: 'bold',
    atom: 'quote',
    em: 'italic',
    quote: 'quote',
    strikethrough: 'strikethrough',
    'variable-2': /^\s*\d+\.\s/.test(line) ? 'ordered-list' : 'unordered-list',
    comment: 'code',
    link: 'link',
    tag: 'image',
    'header-1': 'heading-1',
    'header-2': 'heading-2'
  }

  if (!token.type) {
    return []
  }

  return token.type
    .split(' ')
    .map(type => CODE_MIRROR_TYPES[type])
    .filter(type => type)
}
