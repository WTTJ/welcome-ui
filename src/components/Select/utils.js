export const getSpacer = options =>
  options.reduce(
    (prev, current) => (current.label && prev.length > current.label.length ? prev : current.label),
    ''
  )
