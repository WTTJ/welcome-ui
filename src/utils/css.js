export function getTransformStyle(transform) {
  return {
    transform,
    MozTransform: transform,
    WebkitTransform: transform,
    OTransform: transform,
    MsTransform: transform
  }
}
