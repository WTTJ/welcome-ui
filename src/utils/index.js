export * from './base'
export * from './events'
export * from './fields'
export * from './font'
export * from './getMax'
export * from './hexToRGB'
export * from './propTypes'
export * from './system'
export * from './text'
export * from './validations'
export * from './variants'
