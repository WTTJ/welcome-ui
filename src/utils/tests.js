/* eslint-disable react/prop-types */
import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@xstyled/styled-components'
import { StaticRouter } from 'react-router-dom'
import 'jest-dom/extend-expect'
import 'jest-styled-components'

import { createTheme } from '../../packages/Core/theme/core'
const theme = createTheme()

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <StaticRouter context={{}}>{children}</StaticRouter>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

// https://stackoverflow.com/questions/42099385/jest-enzyme-jsdom-document-body-createtextrange-is-not-a-function
global.document.createRange = () => ({
  setEnd: () => {},
  setStart: () => {},
  getBoundingClientRect: () => ({
    width: 100,
    height: 100,
    top: 0,
    left: 0,
    right: 100,
    bottom: 100
  }),
  getClientRects: () => ({ width: 100, height: 100, top: 0, left: 0, right: 100, bottom: 100 })
})
