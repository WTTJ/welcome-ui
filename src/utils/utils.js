import { css } from 'styled-components'
import {
  alignContent,
  alignItems,
  alignSelf,
  background,
  backgroundColor,
  borders,
  color,
  compose,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontSize,
  fontWeight,
  height,
  justifyContent,
  justifySelf,
  lineHeight,
  order,
  space,
  textAlign,
  width
} from 'styled-system'

export const system = compose(
  alignContent,
  alignItems,
  alignSelf,
  background,
  backgroundColor,
  borders,
  color,
  display,
  flex,
  flexBasis,
  flexDirection,
  flexWrap,
  fontSize,
  fontWeight,
  height,
  justifyContent,
  justifySelf,
  lineHeight,
  order,
  space,
  textAlign,
  width
)

export const overflowEllipsis = css`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`
