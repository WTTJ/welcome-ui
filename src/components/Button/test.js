import React from 'react'

import { render } from '../../utils/tests'

import { Button } from './index'

const content = 'Test button'

test('<Button> renders correctly and has white text by default', () => {
  const button = render(<Button>{content}</Button>).toJSON()
  expect(button).toMatchSnapshot()
  expect(button.children).toContain('Test button')
  expect(button).toHaveStyleRule('color', '#FAFAFA')
  expect(button).toHaveStyleRule('background', '#1B74B5')
})

test('<Button> has correct colour for variant primary', () => {
  const button = render(<Button variant="primary">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#FAFAFA')
  expect(button).toHaveStyleRule('background', '#1B74B5')
})

test('<Button> has correct colour for variant secondary', () => {
  const button = render(<Button variant="secondary">{content}</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#1B74B5')
  expect(button).toHaveStyleRule('background', '#FAFAFA')
})

test('<Button> has correct colour for variant tertiary', () => {
  const button = render(<Button variant="tertiary">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#FAFAFA')
  expect(button).toHaveStyleRule('background', '#333333')
})

test('<Button> has correct colour for variant disabled', () => {
  const button = render(<Button variant="disabled">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#A0A0A0')
  expect(button).toHaveStyleRule('background', '#CCCCCC')
})

test('<Button> has correct colour for variant primary-danger', () => {
  const button = render(<Button variant="primary-danger">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#FAFAFA')
  expect(button).toHaveStyleRule('background', '#D62327')
})

test('<Button> has correct colour for variant secondary-danger', () => {
  const button = render(<Button variant="secondary-danger">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#D62327')
  expect(button).toHaveStyleRule('background', '#FAFAFA')
})

test('<Button> has correct colour for variant primary-warning', () => {
  const button = render(<Button variant="primary-warning">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#FAFAFA')
  expect(button).toHaveStyleRule('background', '#D6772F')
})

test('<Button> has correct colour for variant secondary-warning', () => {
  const button = render(<Button variant="secondary-warning">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#D6772F')
  expect(button).toHaveStyleRule('background', '#FAFAFA')
})
