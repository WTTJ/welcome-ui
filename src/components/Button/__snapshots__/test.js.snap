// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly and has white text by default 1`] = `
.c0 {
  color: #FAFAFA;
  background: #1B74B5;
  border-color: #1B74B5;
  border-radius: 4px;
  font-family: inherit;
  font-size: 0.6875rem;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
  border-width: 1px;
  border-style: solid;
  box-shadow: 0 2px 4px rgba(0,0,0,.2);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition: all .2s cubic-bezier(0.41,0.094,0.54,0.07);
  transition: all .2s cubic-bezier(0.41,0.094,0.54,0.07);
  line-height: 1rem;
  padding: 0.625rem 0.75rem;
}

.c0:active {
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
  box-shadow: 0 2px 4px rgba(0,0,0,.2);
}

.c0::before {
  background: rgba(250,250,250,0.05);
}

.c0[disabled] {
  color: #FAFAFA;
  background-color: #CCCCCC;
  pointer-events: none;
}

@media (max-width:670px) {
  .c0:hover {
    -webkit-transform: translateY(2px);
    -ms-transform: translateY(2px);
    transform: translateY(2px);
  }
}

@media (max-width:670px) {
  .c0 {
    padding: 0.9375rem;
  }
}

<button
  className="c0"
  size="auto"
>
  Test button
</button>
`;
